orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTaskSpec.groovy

```diff
@@ -146,15 +146,65 @@ class MonitorKatoTaskSpec extends Specification {
     "about to fail"      | MonitorKatoTask.TASK_NOT_FOUND_TIMEOUT     |  true         | 'N/A'
   }
 
+  def "should set kato.task.skipReplica=true when getting a task from clouddriver times out"() {
+    given:
+    def taskId = "katoTaskId"
+    def elapsed = MonitorKatoTask.TASK_NOT_FOUND_TIMEOUT + 1
+    task.kato = Mock(KatoService)
+
+    def stage = new Stage(Execution.newPipeline("orca"), "whatever", [
+      "kato.last.task.id": new TaskId(taskId),
+      "kato.task.firstNotFoundRetry": now.minusMillis(elapsed).toEpochMilli()
+    ])
+
+    assert !(stage.context['kato.task.skipReplica'] as Boolean)
+
+    when:
+    def result = task.execute(stage)
+
+    then:
+    result.context['kato.task.skipReplica'] as Boolean
```

Comment:
this is maybe a little more readable as `result.context['kato.task.skipReplica'] == true`

IntelliJ might complain but I still find it more readable